gridstatus/ieso.py

```diff
@@ -2003,42 +2006,145 @@ def get_lmp_day_ahead_hourly(
+        else:
+            files_and_times = self._get_directory_files_and_timestamps(
+                file_directory,
+                file_name_prefix=f"PUB_{directory_path}",
+            )
+
+            # If no end provided, use a default of 1 hour
+            end = end or (date + pd.Timedelta(hours=1))
+
+            urls = [
+                f"{file_directory}/{file}"
+                for file, file_time in files_and_times
+                if date <= file_time < end
+            ]
+
+        if not urls:
+            raise NoDataFoundException(
+                f"No Predispatch Hourly LMP data found for {date} to {end}",
+            )
+
+        data_list = []
+
+        for url in tqdm.tqdm(urls):
+            # We need to get the file created date from the first line of the csv
+            # Example: CREATED AT 2025/05/01 23:14:53 FOR 2025/05/02
+            text = self._request(url, verbose=False).text
+            first_line = text.splitlines()[0]
+
+            match = re.search(
+                r"CREATED AT (\d{4}/\d{2}/\d{2} \d{2}:\d{2}:\d{2})",
+                first_line,
+            )
+
+            publish_timestamp_str = match.group(1)
+
+            publish_time = pd.Timestamp(
+                publish_timestamp_str,
+                tz=self.default_timezone,
+            )
+
+            # Now get the date the file is FOR to use  as the base date
+            match = re.search(r"FOR (\d{4}/\d{2}/\d{2})", first_line)
+            delivery_date = pd.Timestamp(match.group(1), tz=self.default_timezone)
+
+            file_data = self._get_lmp_csv_data(
+                url,
+                base_date=delivery_date,
+                minutes_per_interval=60,
+                verbose=verbose,
+            )
+
+            file_data["Publish Time"] = publish_time
+            data_list.append(file_data)
+
+        data = pd.concat(data_list)
+
+        data = (
+            # It's possible we may have duplicates since some of the files are the same.
+            # We remove these by dropping duplicate rows based on a subset
+            data.drop_duplicates(
+                subset=["Interval Start", "Location", "Publish Time"],
+            )
+        )
+
+        data = (
+            utils.move_cols_to_front(
+                data,
+                [
+                    "Interval Start",
+                    "Interval End",
+                    "Publish Time",
+                    "Location",
+                ],
+            )
+            .sort_values(
+                ["Interval Start", "Location", "Publish Time"],
+            )
+            .reset_index(drop=True)
+        )
+
+        # Remove :LMP from the location
+        data["Location"] = data["Location"].str.replace(":LMP", "")
+
+        return data
+
+    def _get_lmp_csv_data(
         self,
         url: str,
-        date: pd.Timestamp,
-        interval_duration: str = "hour",
+        base_date: pd.Timestamp,
         minutes_per_interval: int = 60,
```

Comment:
since only valid values are 5 and 60 I would probably call this a
```suggestion
        minutes_per_interval: Literal[5, 60] = 60,
```

Reply:
Good idea, updated.